### PR TITLE
use text files for Probeset

### PR DESCRIPTION
**Pull request by Alexanderlacuna**
GN1 uses text files for huge probeset dataset contains strain values.This is an implementation for gn2

### CODE REVIEW

**Comment by Alexanderlacuna** [comment]
@zsloan  could you try this out and give feedback.
@fredmanglis   @BonfaceKilz  this is ready for review

**Review by fredmanglis — APPROVED** [review]
Looks good. LGTM

*nitpick*: Do 

```python
if results:
    ...
```
rather than
```python
if (results):
    ...
```
unless the conditional is particularly long and needs to be broken into multiple lines.

**Review by fredmanglis — APPROVED** [review]
commit 9ee7efd8b18d8caece9cf8d3cc1d58ecbcf209a6

LGTM

**Review by fredmanglis — APPROVED** [review]
commit d6c8505606b7b9fa1e41b3ee8bca303a0f8d6597

LGTM

**Review by fredmanglis — CHANGES_REQUESTED** [review]
commit 6652c3137ec477e02c8291ee6154f24cc3a4cef2

@Alexanderlacuna 

You could use Python's inbuilt [CSV module](https://docs.python.org/3.9/library/csv.html) to parse and dump csv files.

I notice that `parse_line_csv` <sup>1</sup> splits the line into separate values, only for you to [then rejoin them](https://github.com/genenetwork/genenetwork2/pull/727/commits/6652c3137ec477e02c8291ee6154f24cc3a4cef2#diff-47cc07d4d9855d77c1030d0849ba82b992017d7c6bdf14d04d824e121f53cb14R215-R216). What is the point of that? You are doing an operation only to undo it - this is a performance drag (probably not big, but still a performance drag).

The `parse_line_csv` <sup>1</sup> could probably be cleaned up to be clearer on what is happening. The code mutates the `return_list` in a way that is not immediately apparent. It seems to be processing a list of lists, yet parsing a CSV file line would mostly give you non-nested list.

I'm still confused about how this function works. Could you give an example.

I find it difficult to merge this without understanding it. Maybe add a test with a sample file to be read so as to verify it works as expected.

### Footnotes

1 [`parse_line_csv`](https://github.com/genenetwork/genenetwork2/pull/727/commits/6652c3137ec477e02c8291ee6154f24cc3a4cef2#diff-47cc07d4d9855d77c1030d0849ba82b992017d7c6bdf14d04d824e121f53cb14R191-R195)

**Comment by fredmanglis** [comment]
@Alexanderlacuna Please see the review comments and make the appropriate changes and resubmit for review.

File: wqflask/wqflask/correlation/pre_computes.py [hunk 1]
```diff
@@ -167,3 +168,49 @@ def get_datasets_data(base_dataset, target_dataset_data):
...
 
     return (target_results, base_results)
+
+
+def fetch_text_file(dataset_name, conn, text_dir=TEXTDIR):
+    """fetch textfiles with strain vals if exists"""
+
+    with conn.cursor() as cursor:
+        query = 'SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = "%s"' % dataset_name
+        cursor.execute(query)
```
Comment (BonfaceKilz) [inline wqflask/wqflask/correlation/pre_computes.py]: **tip:** Notice you only use query once.  Thus just pass it directly to the execute method.

**issue:** Please use %s like so:
```suggestion
        cursor.execute("SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = %s", (dataset_name,))
```

This sanitizes the string for you and prevents and sql injection from any malicious source.

File: wqflask/wqflask/correlation/pre_computes.py [hunk 2]
```diff
@@ -167,3 +168,49 @@ def get_datasets_data(base_dataset, target_dataset_data):
...
+                _vals.append(target_ids[strain])
+                _posit.append(idx)
+
+        return (_posit, _vals)
+    with open(file_path, "r") as file_handler:
+        all_ids = file_handler.readline()
+        _posit, sample_vals = __fetch_id_positions__(
+            parse_line_csv(all_ids)[1:], sample_dict)
+        return (sample_vals, [filter_line_with_index(line, _posit)
+                              for line in file_handler.readlines()])
```
Comment (BonfaceKilz) [inline wqflask/wqflask/correlation/pre_computes.py]: **issue:** As @fredmanglis mentioned, use python's inbuilt csv module to parse csv files.

**Review by BonfaceKilz — CHANGES_REQUESTED** [review]

**Comment by Alexanderlacuna** [comment]
> commit [6652c31](https://github.com/genenetwork/genenetwork2/commit/6652c3137ec477e02c8291ee6154f24cc3a4cef2)
> 
> @Alexanderlacuna
> 
> You could use Python's inbuilt [CSV module](https://docs.python.org/3.9/library/csv.html) to parse and dump csv files.
> 
> I notice that `parse_line_csv` 1 splits the line into separate values, only for you to [then rejoin them](https://github.com/genenetwork/genenetwork2/pull/727/commits/6652c3137ec477e02c8291ee6154f24cc3a4cef2#diff-47cc07d4d9855d77c1030d0849ba82b992017d7c6bdf14d04d824e121f53cb14R215-R216). What is the point of that? You are doing an operation only to undo it - this is a performance drag (probably not big, but still a performance drag).
> 
> The `parse_line_csv` 1 could probably be cleaned up to be clearer on what is happening. The code mutates the `return_list` in a way that is not immediately apparent. It seems to be processing a list of lists, yet parsing a CSV file line would mostly give you non-nested list.
> 
> I'm still confused about how this function works. Could you give an example.
> 
> I find it difficult to merge this without understanding it. Maybe add a test with a sample file to be read so as to verify it works as expected.
> 
> ### Footnotes
> 1 [`parse_line_csv`](https://github.com/genenetwork/genenetwork2/pull/727/commits/6652c3137ec477e02c8291ee6154f24cc3a4cef2#diff-47cc07d4d9855d77c1030d0849ba82b992017d7c6bdf14d04d824e121f53cb14R191-R195)

@fredmanglis  This has to do with the bad formatting  of the current text files being used and makes it impossible to parse with as csv for example
the values for each are quoted as shown below
'"1415918_a_at","13.948","14.247","13.965","14.298","14.137","14.288","14.186","13.988","14.156","14.009","14.259","13.861","14.029"'  to have
['1415918_a_at', '13.948', '14.247', '13.965', '14.298', '14.137', '14.288', '14.186', '13.988', '14.156', '14.009', '14.259', '13.861', '14.02']
then perform a join to be in the required format
this should change once we generate our own text files
You can try to parse such a file (below)  to  get a better understanding.
[text_files.zip](https://github.com/genenetwork/genenetwork2/files/9507677/text_files.zip)

**Comment by Alexanderlacuna** [comment]
@fredmanglis  PS:I tempted this with csv and it does a better job,thanks for this

**Comment by fredmanglis** [comment]
> > commit [6652c31](https://github.com/genenetwork/genenetwork2/commit/6652c3137ec477e02c8291ee6154f24cc3a4cef2)
> > @Alexanderlacuna
> > You could use Python's inbuilt [CSV module](https://docs.python.org/3.9/library/csv.html) to parse and dump csv files.
> > I notice that `parse_line_csv` 1 splits the line into separate values, only for you to [then rejoin them](https://github.com/genenetwork/genenetwork2/pull/727/commits/6652c3137ec477e02c8291ee6154f24cc3a4cef2#diff-47cc07d4d9855d77c1030d0849ba82b992017d7c6bdf14d04d824e121f53cb14R215-R216). What is the point of that? You are doing an operation only to undo it - this is a performance drag (probably not big, but still a performance drag).
> > The `parse_line_csv` 1 could probably be cleaned up to be clearer on what is happening. The code mutates the `return_list` in a way that is not immediately apparent. It seems to be processing a list of lists, yet parsing a CSV file line would mostly give you non-nested list.
> > I'm still confused about how this function works. Could you give an example.
> > I find it difficult to merge this without understanding it. Maybe add a test with a sample file to be read so as to verify it works as expected.
> > ### Footnotes
> > 1 [`parse_line_csv`](https://github.com/genenetwork/genenetwork2/pull/727/commits/6652c3137ec477e02c8291ee6154f24cc3a4cef2#diff-47cc07d4d9855d77c1030d0849ba82b992017d7c6bdf14d04d824e121f53cb14R191-R195)
> 
> @fredmanglis This has to do with the bad formatting of the current text files being used and makes it impossible to parse with as csv for example the values for each are quoted as shown below '"1415918_a_at","13.948","14.247","13.965","14.298","14.137","14.288","14.186","13.988","14.156","14.009","14.259","13.861","14.029"' to have ['1415918_a_at', '13.948', '14.247', '13.965', '14.298', '14.137', '14.288', '14.186', '13.988', '14.156', '14.009', '14.259', '13.861', '14.02'] then perform a join to be in the required format this should change once we generate our own text files You can try to parse such a file (below) to get a better understanding. [text_files.zip](https://github.com/genenetwork/genenetwork2/files/9507677/text_files.zip)

I just tested this with the file you provided with  [test_read.py.zip](https://github.com/genenetwork/genenetwork2/files/9525901/test_read.py.zip) (extract it in the `text_files` directory) and csv reads the file with no issues as far as I can tell.

Please have a look and get back to me.

File: wqflask/wqflask/correlation/pre_computes.py [hunk 3]
```diff
@@ -174,10 +175,10 @@ def fetch_text_file(dataset_name, conn, text_dir=TEXTDIR):
     """fetch textfiles with strain vals if exists"""
 
     with conn.cursor() as cursor:
-        query = 'SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = "%s"' % dataset_name
-        cursor.execute(query)
+        cursor.execute(
+            'SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = "%s"' % dataset_name)
```
Comment (fredmanglis) [inline wqflask/wqflask/correlation/pre_computes.py]: Change `"%s"` to `%s` (no double quotes) and don't do string interpolation.

Change the entire thing to:

```python
cursor.execute(
    'SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = %s', dataset_name)
```

**NOTE**: the text of the query (`"SELECT  Id, ..."`) is separated from the parameter (`dataset_name`) with a comma (**,**) not a string interpolation operator (**%**)

**Review by fredmanglis — CHANGES_REQUESTED** [review]
Change `"%s"` to `%s` (no double quotes) and don't do string interpolation.

Change the entire thing to:

```python
cursor.execute(
    'SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = %s', dataset_name)
```

**NOTE**: the text of the query (`"SELECT  Id, ..."`) is separated from the parameter (`dataset_name`) with a comma (**,**) not a string interpolation operator (**%**)

**Pull request merged**